File: scratchpad/7. Reverse Integer.cpp
#include <bits/stdc++.h>

class Solution {
public:
    int reverse(int x) {
        int temp = abs(x); 
            
        long rev = 0;
        while (temp != 0) {
            rev = rev * 10 + (temp % 10); 
            temp /= 10;
        }

        if (x < 0) rev *= -1;

        if (!((rev >= INT_MIN) && (rev <= INT_MAX))) return 0;

        return rev;
    }
};

int main() {
    std::cout << (new Solution())->reverse(1234) << "\n";
}

File: numbers/7. Reverse Integer.py
import math

class Solution:
    def rev(self, num: str) -> int:
        return int(num[::-1])

    def reverse(self, x: int) -> int:
        num = self.rev(str(x) if x > 0 else str(-x))
        num = -num if x < 0 else num

        return 0 if not (num < math.pow(2, 31) - 1 and num > math.pow(-2, 31)) else num